### ISSUE TITLE
SIGSEGV in System.Numerics.Vectors

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/runtime#94856]
Saw this here: https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-94756-merge-dc464ce7a6094aecb0/System.Numerics.Vectors.Tests/1/console.d55d86b1.log?helixlogtype=result

```
/root/helix/work/workitem/e /root/helix/work/workitem/e
Running assembly:System.Numerics.Vectors.Tests, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51
./RunTests.sh: line 204:    23 Segmentation fault      (core dumped) ./System.Numerics.Vectors.Tests -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing -xml testResults.xml $RSP_FILE
/root/helix/work/workitem/e
----- end Thu 16 Nov 2023 03:38:07 AM UTC ----- exit code 139 ----------------------------------------------------------
exit code 139 means SIGSEGV Illegal memory access. Deref invalid pointer, overrunning buffer, stack overflow etc. Core dumped.
```

I pulled down the dump, the stack looks like this:

```
00 00007f00`3e9d9378 00007f40`d401e140     System_Numerics_Vectors!SIGSEGVHandler
01 00007f00`3e9d9380 00005616`bfd52584     libpthread_2_31!funlockfile+0x50
02 00007f00`36ffb520 00005616`bfbff1a9     System_Numerics_Vectors!S_P_CoreLib_System_Numerics_Vector_1<Int8>::CopyTo+0x4 [/_/src/libraries/System.Private.CoreLib/src/System/Numerics/Vector_1.cs @ 681] 
03 00007f00`36ffb530 00005616`bff93f79     System_Numerics_Vectors!xunit_assert_Xunit_Assert::RecordException+0x29 [/_/src/Microsoft.DotNet.XUnitAssert/src/Record.cs @ 37] 
04 00007f00`36ffb550 00005616`bff02f34     System_Numerics_Vectors!xunit_assert_Xunit_Assert::Throws+0x29 [/_/src/Microsoft.DotNet.XUnitAssert/src/ExceptionAsserts.cs @ 100] 
05 00007f00`36ffb580 00005616`bffac207     System_Numerics_Vectors!System_Numerics_Vectors_Tests_System_Numerics_Tests_GenericVectorTests::TestCopyTo+0x94 [/_/src/libraries/System.Numerics.Vectors/tests/GenericVectorTests.cs @ 638] 
06 00007f00`36ffb5f0 00005616`bfad81a9     System_Numerics_Vectors!Internal_CompilerGenerated__Module___<DynamicInvoke>None<S_P_CoreLib_System_Void>+0x17 [/_/src/libraries/System.Linq/src/System/Linq/Join.cs @ 64] 
07 00007f00`36ffb610 00005616`bfb65c6f     System_Numerics_Vectors!S_P_CoreLib_System_Reflection_DynamicInvokeInfo::Invoke+0xa9 [/_/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/DynamicInvokeInfo.cs @ 228] 
```

The test expects throwing a NullRef and I think we hit it, but looks like we didn't translate it?

https://github.com/dotnet/runtime/blob/0b073c37f99c17fde1dbf8e379b1c3dcfa9c4f38/src/libraries/System.Numerics.Vectors/tests/GenericVectorTests.cs#L637

**Comment by jkotas** [comment]
I have seen crashes with similar symptoms on other PRs. I think it may be a regression introduced by #94485.

cc @jtschuster

**Comment by jkotas** [comment]
Fixed by #95415
